src/widgets/date-picker/triggerPanel/rangePanelTrigger.js

```diff
@@ -237,20 +237,23 @@ class Range extends Component<TypeProps, TypeState> {
       if (onOk || showTime) {
         visible = true;
       }
-      const { isValid } = this.getIsValid(renderValue);
-      const { onChange } = this.props;
-      const sortValue = this.getSortValue(renderValue, format);
-      isValid &&
-        onChange &&
-        onChange({ newValue: sortValue, oldValue: this.changeOldValue, event });
+      const sortValue = this.exportOnChange(renderValue, this.changeOldValue, event);
       setStateData = { value: sortValue, rangeValue: [], isHover: false };
-      if (isValid) {
-        this.onBlur();
-      }
     }
     this.drawPageAgain(renderValue, format);
     this.setState({ ...setStateData, visible });
   };
+  exportOnChange = (renderValue: string[], oldValue: string[], event: Object) => {
+    const { format } = this.state;
+    const { isValid } = this.getIsValid(renderValue);
+    const { onChange } = this.props;
+    const newValue = this.getSortValue(renderValue, format);
+    isValid && onChange && onChange({ newValue, oldValue, event });
+    if (isValid) {
+      this.onBlur();
+    }
```

Comment:
if (isValid) {
      onChange && onChange({ newValue, oldValue, event })
      this.onBlur();
    }